### PR TITLE
Ecsra 47 get endpoint to viewing cart

### PR DESCRIPTION
**Pull request by MariiaLobanova**
link to jira board - https://mariialobanova.atlassian.net/jira/software/c/projects/ECSRA/boards/2?selectedIssue=ECSRA-47

Added GET endpoint that allow user to see his cart, added EmptyCartErrorHandler,  fix typo in exception-package. Endpoint tested in postman sussefully: 
Image 30 01 24 at 11 57
Image 30 01 24 at 12 28

### CODE REVIEW

File: src/main/java/startsteps/ECommerceShop/service/CartService.java [hunk 1]
```diff
@@ -98,4 +96,29 @@ private List<CartProductResponse> createCartProduct(List<CartProduct> cartProduc
...
+        }
+        log.info("get cart: {}", user.getCart());
+
+        List<CartProductResponse> cartProductResponses = cart.getCartProductList().stream()
+                .map(this::mapToCartProductResponse).collect(Collectors.toList());
+
+        return new CartResponse("Cart details retrieved successfully", cartProductResponses, cart.getTotalPrice());
+    }
+
+    private CartProductResponse mapToCartProductResponse(CartProduct cartProduct){
```
Comment (faruka-startsteps) [inline src/main/java/startsteps/ECommerceShop/service/CartService.java:115]: you can have a constructor in CartProductResponse that takes in one argument of Product and construct the object that way

File: src/main/java/startsteps/ECommerceShop/service/CartService.java [hunk 2]
```diff
@@ -98,4 +96,29 @@ private List<CartProductResponse> createCartProduct(List<CartProduct> cartProduc
...
         }
+
+    @Transactional
+    public CartResponse getCart(User user){
+        Cart cart = user.getCart();
+
+        if (cart == null || cart.getCartProductList().isEmpty()){
+            throw new EmptyCartException("You haven't added anything yet, your cart is empty");
+        }
+        log.info("get cart: {}", user.getCart());
```
Comment (faruka-startsteps) [inline src/main/java/startsteps/ECommerceShop/service/CartService.java:107]: you can pass in `cart` that you defined on line 102

File: src/main/java/startsteps/ECommerceShop/exceptions/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -59,4 +57,14 @@ public ResponseEntity<Object> handleRuntimeException(RuntimeException exception)
                 .status(HttpStatus.INTERNAL_SERVER_ERROR)
                 .body(errorResponse);
     }
+    @ExceptionHandler(EmptyCartException.class)
+    public ResponseEntity<Object> handleEmptyCartException(EmptyCartException exception) {
+        ErrorResponse errorResponse = new ErrorResponse(
+                HttpStatus.NOT_FOUND.value(),
```
Comment (faruka-startsteps) [inline src/main/java/startsteps/ECommerceShop/exceptions/GlobalExceptionHandler.java:63]: I wouldn't return 404 if the cart is empty. Maybe 200 with empty data or 204 with no data

File: src/main/java/startsteps/ECommerceShop/controller/CartController.java [hunk 4]
```diff
@@ -37,4 +35,12 @@ public ResponseEntity<CartResponse> addToCart(@RequestBody CartRequest cartReque
 
         return ResponseEntity.ok(cartResponse);
     }
+
+    @GetMapping("/mycart")
+    @PreAuthorize("hasAuthority('USER')")
+   public CartResponse getCart(User user){
+        User user1 = authService.getAuthenticatedUser();
```
Comment (faruka-startsteps) [inline src/main/java/startsteps/ECommerceShop/controller/CartController.java:42]: why user two times

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/main/java/startsteps/ECommerceShop/service/CartService.java (hunk 1 above)
Comment (MariiaLobanova) [reply src/main/java/startsteps/ECommerceShop/service/CartService.java:115]: done, implemented a constructor.

File: src/main/java/startsteps/ECommerceShop/exceptions/GlobalExceptionHandler.java (hunk 3 above)
Comment (MariiaLobanova) [reply src/main/java/startsteps/ECommerceShop/exceptions/GlobalExceptionHandler.java:63]: deleted exception, created new response with message, empty list and total price = 0

File: src/main/java/startsteps/ECommerceShop/service/CartService.java (hunk 2 above)
Comment (MariiaLobanova) [reply src/main/java/startsteps/ECommerceShop/service/CartService.java:107]: done

File: src/main/java/startsteps/ECommerceShop/service/CartService.java (hunk 1 above)
Comment (f4r33) [reply src/main/java/startsteps/ECommerceShop/service/CartService.java:113]: you can delete this function now

**Review by f4r33 — APPROVED** [review]

File: src/main/java/startsteps/ECommerceShop/service/CartService.java (hunk 1 above)
Comment (MariiaLobanova) [reply src/main/java/startsteps/ECommerceShop/service/CartService.java]: done

**Review by faruka-startsteps — APPROVED** [review]

**Pull request merged**